packages/block-editor/src/components/provider/with-registry-provider.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect } from '@wordpress/element';
+import { withRegistry, createRegistry, RegistryProvider } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { storeConfig } from '../../store';
+import applyMiddlewares from '../../store/middlewares';
+
+const withRegistryProvider = createHigherOrderComponent( ( WrappedComponent ) => {
+	return withRegistry( ( { useSubRegistry = true, registry, ...props } ) => {
+		// Disable reason, this rule conflicts with the React hooks rule (no conditionals)
+		// eslint-disable-next-line @wordpress/no-unused-vars-before-return
+		const [ subRegistry, updateRegistry ] = useState( null );
```

Comment:
Well, this is strange. Why bother and call it if it is not going to be used? `useEffect` is put after the if statement which might terminate the execution of this render method. I think it should be perfectly fine to put `useState` usage just before `useEffect`.

After React docs (https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level):

> **Don’t call Hooks inside loops, conditions, or nested functions.** Instead, always use Hooks at the top level of your React function. 

This is going to be satisfied.

Reply:
mmm right! I think I misunderstood the rules.